packages/sitecore-jss-react/src/components/FEaaSComponent.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect } from 'react';
+import { ComponentParams, ComponentFields, getFieldValue } from '@sitecore-jss/sitecore-jss/layout';
+
+export const FEAAS_COMPONENT_RENDERING_NAME = 'FEaaSComponent';
+
+export type FEaaSComponentParams = ComponentParams & {
+  LibraryId?: string;
+  ComponentId?: string;
+  ComponentVersion?: string;
+  ComponentRevision?: string;
+  ComponentHostName?: string;
+  ComponentInstanceId?: string;
+  ComponentDataOverride?: string;
+  ComponentHTMLOverride?: string;
+};
+
+export type FEaaSComponentProps = {
+  params?: FEaaSComponentParams;
+  fields?: ComponentFields;
+};
+
+const getDataFromFields = (fields: ComponentFields): { [key: string]: unknown } => {
+  let data: { [key: string]: unknown } = {};
+  data = Object.entries(fields).reduce((acc, [key]) => {
+    acc[key] = getFieldValue(fields, key);
+    return acc;
+  }, data);
+  return data;
+};
+
+export const FEaaSComponent = ({ params, fields }: FEaaSComponentProps): JSX.Element => {
+  if (
+    !params ||
+    !params.LibraryId ||
+    !params.ComponentId ||
+    !params.ComponentVersion ||
+    !params.ComponentRevision ||
+    !params.ComponentHostName
+  ) {
+    // Missing FEaaS component required props
+    return null;
```

Comment:
It will not render anything at all in this case. We could continue with rendering the FEaaS web components if you think this would be preferred, but wasn't sure about full implications in this case (e.g. just console errors? or visual issues on the page as well?)

Reply:
Yeah i think that's fine. In general it should not fail and invalidate it in runtime, but i think this approach is good. I think the need for having unconfigured component to render will not arise